src/test/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSinkUnitTest.java

```diff
@@ -64,9 +64,26 @@ public void readsSinkTest(String inputBam, String outputFile) throws IOException
         Assert.assertEquals(rddParallelReads.count(), rddParallelReads2.count());
     }
 
+    @Test(dataProvider = "loadReadsBAM", groups = "spark")
+    public void readsSinkShardedTest(String inputBam, String outputFile) throws IOException {
+        new File(outputFile).delete();
+        JavaSparkContext ctx = SparkContextFactory.getTestSparkContext();
+
+        ReadsSparkSource readSource = new ReadsSparkSource(ctx);
+        JavaRDD<GATKRead> rddParallelReads = readSource.getParallelReads(inputBam);
+        rddParallelReads = rddParallelReads.repartition(2);
```

Comment:
Is this guaranteeing that the output is in 2 shards?  Maybe add a comment here?  An explicit check once it's written to disk that it actually wrote 2 files would be good since I can imagine HadoopBAM could theoretically coalesce internally when the file sie is small.